File: webdriver-framework-core/src/main/java/io/github/webdriver/config/WebDriverConfiguration.java
package io.github.webdriver.config;

import java.util.Collections;
import java.util.List;
import java.util.Map;

import org.apache.commons.lang3.StringUtils;

import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonProperty;

public class WebDriverConfiguration {

    private String testEndpoint;

    private Map<String, String> namedTestUrls;

    private DriverType driverType;

    @JsonProperty("onPremDriver")
    private OnPremDriverConfig onPremDriverConfig;

    @JsonProperty("onPremGridDriver")
    private RemoteDriverConfig onPremGridDriverConfig;

    @JsonProperty("cloudDriver")
    private RemoteDriverConfig cloudDriverConfig;

    public List<Platform> getActivePlatforms() {
        List<Platform> activePlatforms = Collections.emptyList();
        switch (driverType) {
            case onPremDriver:
                activePlatforms = onPremDriverConfig.getPlatforms();
                break;
            case onPremGridDriver:
                activePlatforms = onPremGridDriverConfig.getPlatforms();
                break;
            case cloudDriver:
                activePlatforms = cloudDriverConfig.getPlatforms();
                break;
        }
        return activePlatforms;
    }

    public String getTestEndpoint() {
        return testEndpoint;
    }

    public void setTestEndpoint(String testEndpoint) {
        this.testEndpoint = testEndpoint;
    }

    public DriverType getDriverType() {
        return driverType;
    }

    public void setDriverType(DriverType driverType) {
        this.driverType = driverType;
    }

    public OnPremDriverConfig getOnPremDriverConfig() {
        return onPremDriverConfig;
    }

    public void setOnPremDriverConfig(OnPremDriverConfig onPremDriverConfig) {
        this.onPremDriverConfig = onPremDriverConfig;
    }

    public RemoteDriverConfig getOnPremGridDriverConfig() {
        return onPremGridDriverConfig;
    }

    public void setOnPremGridDriverConfig(RemoteDriverConfig onPremGridDriverConfig) {
        this.onPremGridDriverConfig = onPremGridDriverConfig;
    }

    public RemoteDriverConfig getCloudDriverConfig() {
        return cloudDriverConfig;
    }

    public void setCloudDriverConfig(RemoteDriverConfig cloudDriverConfig) {
        this.cloudDriverConfig = cloudDriverConfig;
    }

    @JsonAnySetter
    public void setNamedTestUrls(String key, String value) {
        this.namedTestUrls.put(key, value);
    }

    public Map<String, String> getNamedTestUrls() {
        return this.namedTestUrls;
    }

    public String getNamedUrl(String name) {
        if (StringUtils.isEmpty(name)) {
            return "";
        }
        return this.namedTestUrls.get(name);
    }
}

File: webdriver-framework-core-python/src/main/webdriver/config/WebDriverConfiguration.py
from config import DriverType
from config import OnPremDriverConfig
from config import RemoteDriverConfig

class WebDriverConfigurationClass:

    __testEndpoint              :str = None
    __namedTestUrls             :dict = {} 
    __driverType                : DriverType.DriverTypeClass = None
    __onPremDriverConfig        : OnPremDriverConfig.OnPremDriverConfigClass = None
    __onPremGridDriverConfig    : RemoteDriverConfig.RemoteDriverConfigClass = None
    __cloudDriverConfig         : RemoteDriverConfig.RemoteDriverConfigClass = None

    def getActivePlatforms(self):
        activePlatforms = []

        if(self.__driverType.name == "onPremDriver" ):
            activePlatforms = self.__onPremDriverConfig.getPlatforms()

        elif(self.__driverType.name == "onPremGridDriver"):
            activePlatforms = self.__onPremGridDriverConfig.getPlatforms()

        elif(self.__driverType.name == "cloudDriver"):
            activePlatforms = self.__cloudDriverConfig.getPlatforms()

        return activePlatforms

    def getTestEndpoint(self):
        return self.__testEndpoint

    def setTestEndpoint(self, testEndpoint :str ):
        self.__testEndpoint = testEndpoint

    def getDriverType(self):
        return self.__driverType.name

    def setDriverType(self, driverType : DriverType.DriverTypeClass):
        self.__driverType = driverType

    def getOnPremDriverConfig(self):
        return self.__onPremDriverConfig

    def setOnPremDriverConfig(self, onPremDriverConfig : OnPremDriverConfig):
        self.__onPremDriverConfig = onPremDriverConfig

    def getOnPremGridDriverConfig(self) :
        return self.__onPremGridDriverConfig

    def setOnPremGridDriverConfig(self, onPremGridDriverConfig: RemoteDriverConfig):
        self.__onPremGridDriverConfig = onPremGridDriverConfig

    def getCloudDriverConfig(self):
        return self.__cloudDriverConfig

    def setCloudDriverConfig(self, cloudDriverConfig: RemoteDriverConfig):
        self.__cloudDriverConfig = cloudDriverConfig

    def setNamedTestUrls(self, key: str, value: str):
        self.__namedTestUrls[key] = value

    def getNamedTestUrls(self):
        return self.__namedTestUrls

    def getNamedUrl(self, name: str):
        if (name == None or name == ""):
            return ""
        return self.__namedTestUrls[name]